tools/ops/aws-iam-gen.py

```diff
@@ -0,0 +1,157 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""
+Tool to generate a list of iam permissions from a
+directory of policies.
+"""
+
+import click
+from collections import defaultdict
+import functools
+import json
+from pathlib import Path
+import yaml
+
+from c7n import resources, config, loader
+from c7n.resources import aws
+
+
+@click.group()
+def cli():
+    """aws iam permissions generation"""
+
+
+@cli.command()
+@click.option("-f", "--iam-file", type=click.File("rb"), multiple=True)
+@click.option("--output", type=click.File("w"), default="-")
+def merge(iam_file, output):
+    """Merge multiple iam permission files"""
+    perms = set()
+    for f in iam_file:
+        perms.update(yaml.safe_load(f.read()))
+    output.write(yaml.dump(sorted(perms), default_flow_style=False))
+
+
+@cli.command()
+@click.option("-d", "--directory", required=True)
+@click.option("--name-glob", multiple=True)
+@click.option("--output", type=click.File("w"), default="-")
+def from_policy(directory, name_glob, output):
+    """Generate an iam permission list from a directory of custodian policy files"""
+    ploader = loader.PolicyLoader(config.Config.empty())
+    pdir = Path(directory)
+    collections = []
+    for f in pdir.rglob("*.yaml"):
+        collections.append(ploader.load_file(f))
+    policies = functools.reduce(lambda x, y: x + y, collections)
```

Comment:
Testing this locally I realized I had a directory full of `.yml` files, so running this threw:

```console
TypeError: reduce() of empty iterable with no initial value
```

It could be worth covering that case with either:

```suggestion
    for f in pdir.rglob("*.y*ml"):
        collections.append(ploader.load_file(f))
    policies = functools.reduce(lambda x, y: x + y, collections)
```

and/or:

```suggestion
    for f in pdir.rglob("*.yaml"):
        collections.append(ploader.load_file(f))
    if not collections:
        print("No policies found")
        return
    policies = functools.reduce(lambda x, y: x + y, collections)
```